src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,52 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserCredentialsException;
+import core.basesyntax.exception.InvalidUserObjectException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (validateUserCredentials(user)) {
```

Comment:
You should add a check to ensure that the user's login is unique in the storage before adding the user. This is a requirement from the task description: 'there is no user with such login in the Storage yet'.